engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoMultiValue.java

```diff
@@ -83,285 +99,264 @@ public final String toDisplayString(boolean ignore) {
     return toString();
   }
 
-  @ExportMessage
-  boolean isBoolean(@Shared("interop") @CachedLibrary(limit = "10") InteropLibrary iop) {
-    for (var i = 0; i < values.length; i++) {
-      if (iop.isBoolean(values[i])) {
-        return true;
+  private enum InteropType {
+    NULL,
+    BOOLEAN,
+    DATE_TIME_ZONE,
+    DURATION,
+    STRING,
+    NUMBER,
+    POINTER,
+    META_OBJECT,
+    ITERATOR;
```

Comment:
Turns out that [these types](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html) are **mutually exclusive**:

> The following type combinations are mutually exclusive and cannot return true for the type check message of the same receiver value:
> [Null](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isNull(java.lang.Object))
> [Boolean](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isBoolean(java.lang.Object))
> [String](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isString(java.lang.Object))
> [Number](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isNumber(java.lang.Object))
> [Date](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isDate(java.lang.Object)), [Time](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isTime(java.lang.Object)) or [TimeZone](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isTimeZone(java.lang.Object))
> [Duration](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isDuration(java.lang.Object))
> [Exception](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isException(java.lang.Object))
> [Meta-Object](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isMetaObject(java.lang.Object))
> [Iterator](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isIterator(java.lang.Object))

as such the following `find` method always selects at most **one of the above types** and that is the _interop type_ of the _multi value_.